From b212a87b7452d3ca607b3089eaf6ac7285c1302b Mon Sep 17 00:00:00 2001
From: Steve Kondik <steve@cyngn.com>
Date: Mon, 19 Oct 2015 14:43:39 -0700
Subject: [PATCH 2/2] power: Add support for tap-to-wake feature control

Squashed into this commit:

Author: LuK1337 <priv.luk@gmail.com>
Date:   Sat Dec 7 23:51:29 2019 +0100

    power: Follow symlinks when writing to dt2w toggle path

    * Some devices tend to symlink touchpanel related toggles
      to common path due to having to deal with different panels.

    Change-Id: I429e4c5acfbcd3ff26a019f919fbe880692160bd

[ArianK16a]: Adapt to AIDL HAL

Change-Id: I545902b29f4828c127bc32def6e30b67ce4a3aa7
---
 Android.mk |  4 ++++
 Power.cpp  | 10 ++++++++++
 2 files changed, 14 insertions(+)

diff --git a/Android.mk b/Android.mk
index 7be7a23..d6967dc 100644
--- a/Android.mk
+++ b/Android.mk
@@ -81,6 +81,10 @@ ifeq ($(call is-board-platform-in-list,msmnile), true)
 LOCAL_SRC_FILES += power-msmnile.c
 endif
 
+ifneq ($(TARGET_TAP_TO_WAKE_NODE),)
+    LOCAL_CFLAGS += -DTAP_TO_WAKE_NODE=\"$(TARGET_TAP_TO_WAKE_NODE)\"
+endif
+
 ifeq ($(TARGET_USES_INTERACTION_BOOST),true)
     LOCAL_CFLAGS += -DINTERACTION_BOOST
 endif
diff --git a/Power.cpp b/Power.cpp
index 260ad4c..a28df32 100644
--- a/Power.cpp
+++ b/Power.cpp
@@ -31,6 +31,7 @@
 
 #include "Power.h"
 
+#include <android-base/file.h>
 #include <android-base/logging.h>
 
 #include <aidl/android/hardware/power/BnPower.h>
@@ -60,7 +61,13 @@ void setInteractive(bool interactive) {
 ndk::ScopedAStatus Power::setMode(Mode type, bool enabled) {
     LOG(INFO) << "Power setMode: " << static_cast<int32_t>(type) << " to: " << enabled;
     switch(type){
+#ifdef TAP_TO_WAKE_NODE
         case Mode::DOUBLE_TAP_TO_WAKE:
+            ::android::base::WriteStringToFile(enabled ? "1" : "0", TAP_TO_WAKE_NODE, true);
+            break;
+#else
+        case Mode::DOUBLE_TAP_TO_WAKE:
+#endif
         case Mode::LOW_POWER:
         case Mode::LAUNCH:
         case Mode::EXPENSIVE_RENDERING:
@@ -93,6 +100,9 @@ ndk::ScopedAStatus Power::isModeSupported(Mode type, bool* _aidl_return) {
     LOG(INFO) << "Power isModeSupported: " << static_cast<int32_t>(type);
 
     switch(type){
+#ifdef TAP_TO_WAKE_NODE
+        case Mode::DOUBLE_TAP_TO_WAKE:
+#endif
         case Mode::INTERACTIVE:
         case Mode::SUSTAINED_PERFORMANCE:
         case Mode::FIXED_PERFORMANCE:
-- 
2.37.1

